src/pages/macros.tsx

```diff
@@ -89,19 +93,41 @@ const Macros = () => {
     );
   };
 
-  useEffect(() => {
+  const fetchAll = async () => {
     if (!selectedInstance) return;
+    fetchMacros(selectedInstance.uuid);
+    fetchTasks(selectedInstance.uuid);
+    fetchHistory(selectedInstance.uuid);
+  };
 
-    const fetchAll = async () => {
-      if (!selectedInstance) return;
-      fetchMacros(selectedInstance.uuid);
-      fetchTasks(selectedInstance.uuid);
-      fetchHistory(selectedInstance.uuid);
-    };
-
+  useEffect(() => {
     fetchAll();
   }, [selectedInstance]);
 
+  const { data: runningMacros } = useQuery([
```

Comment:
I'm a little confused on the difference between runningMacros and taskList/fetchTasks/getTasks, but it might be better if we use `useQuery` to drive the fetch and not call `fetchAll` and `setQueryData` manually.

As in something like
```tsx
const {data: tasks } = useQuery(
	['instance', selectedInstance?.uuid, 'taskList'],
	() => getTasks().map(idk)
	{enabled: selectedInstance != null});
```